### PR TITLE
Test of a simple object mapping

### PR DESCRIPTION
**Pull request by noplay**
It's a suggestion of improvements with a basic object mapping

### CODE REVIEW

**Comment by ehlers** [comment]
Thanks a lot for your suggestion, I really appreciate it.

Just to clarify:
I haven't done much python programming, so this repository wasn't meant as the main repository for a python GNS3 API module. I think, that will exceed my python skills.

It was more meant as an inspiration for "official" API modules hosted/maintained by GNS3. That's what https://github.com/GNS3/gns3-gui/issues/1847 was meant for.

**Comment by noplay** [comment]
Yeah I understand. I see it as an experimentation for a future python module.

**Comment by ehlers** [comment]
Ok, that's fine for me. Let's play...

**Comment by Raizo62** [comment]
Hi

Can we use this api (or  [api.gns3.net/en/2.1](http://api.gns3.net/en/2.1/)) to **configure** new VM (Vbox, Qemu, ...) on GNS3 ? (I don't want to add devices to topology, but really in configuration)

**Comment by Raizo62** [comment]
I believe that i succeeded to change the config of gns3_controlleur (http://localhost:3080/v2/settings").

But, gns3-gui doesn't know the changes.

Can i say to gns3-gui to update his config ? i can't ? It is impossible because i don't understood something ?

**Comment by Raizo62** [comment]
We can. I found how. Thanks :-D

**Comment by noplay** [comment]
:)

File: gns3api.py [hunk 1]
```diff
@@ -11,6 +11,37 @@
 from base64 import b64encode
 
 
+class Element:
+    def __init__(self, http, data):
```
Comment (ehlers) [inline gns3api.py]: Just had a closer look.
Isn't the argument http a bit misleading? Currently it is the object of the class, within that Element object is created. So for Project objects it is an api object, for Node objects it's a Project object. While for Project objects the name http is fine, for Node objects it's strange. Maybe something more neutral is a better name.

Or do you really want an api object? Then the class creation for Node should be something like "Node(self._http, node)".

Or have I overlooked something?

**Comment by noplay** [comment]
Oh yes it was a mistake we should drop the http

Le mer. 15 févr. 2017 à 02:56, Bernhard Ehlers <notifications@github.com> a
écrit :

**Comment by ehlers** [comment]
I'm puzzled. When we drop the http parameter, how can a Project object create an api.request call?

**Comment by noplay** [comment]
Oh my bad. We should rename it

Le mer. 15 févr. 2017 à 10:11, Bernhard Ehlers <notifications@github.com> a
écrit :

**Comment by ehlers** [comment]
I still don't know, what exactly you want. So I will revert the object mapping.

**Comment by noplay** [comment]
We need access to the api in the node object. Ideally it could be named api but the name is already use.

Perhaps it should be api and:

class api: should be rename to class Api: or  class GNS3Api:

**Pull request merged**